js/src/builder.ts

```diff
@@ -289,7 +290,7 @@ export abstract class Builder<T extends DataType = any, TNull = any> {
             valueOffsets = _offsets?.flush(length);
         } else if (valueOffsets = _offsets?.flush(length)) { // Variable-width primitives (Binary, Utf8), and Lists
             // Binary, Utf8
-            data = _values?.flush(_offsets.last());
+            data = _values?.flush(Number(_offsets.last()));
```

Comment:
I think this will fail for very large offsets that are larger than `number` allows but I think that's inevitable.

Reply:
Inevitable because JS supports indexing with numbers only.